pkgs/os-specific/linux/intel-sgx/default.nix

```
@@ -0,0 +1,296 @@
+      cp ${build}/le_prod_css.bin $out/aesm/
+      cp ${build}/libsgx_le.signed.so $out/aesm/
+      cp ${build}/libsgx_pve.signed.so $out/aesm/
+      cp ${build}/libsgx_qe.signed.so $out/aesm/
+      cp ${build}/libsgx_pce.signed.so $out/aesm/
+
+      cp external/dcap_source/QuoteGeneration/psw/ae/data/prebuilt/libsgx_qe3.signed.so $out/aesm/
+      cp external/dcap_source/QuoteGeneration/build/linux/libsgx_pce_logic.so $out/aesm/
+      cp external/dcap_source/QuoteGeneration/build/linux/libsgx_qe3_logic.so $out/aesm/
+
+      # copy bundle binaries
+      mkdir -p $out/aesm/bundles
+      cp ${build}/bundles/libepid_quote_service_bundle.so $out/aesm/bundles/
+      cp ${build}/bundles/lible_launch_service_bundle.so $out/aesm/bundles/
+      cp ${build}/bundles/liblinux_network_service_bundle.so $out/aesm/bundles/
+      cp ${build}/bundles/libpce_service_bundle.so $out/aesm/bundles/
+      cp ${build}/bundles/libepid_quote_service_bundle.so $out/aesm/bundles/
+      cp ${build}/bundles/libecdsa_quote_service_bundle.so $out/aesm/bundles/
+      cp ${build}/bundles/libquote_ex_service_bundle.so $out/aesm/bundles/
+
+      mkdir -p $out/aesm/data
+      cp psw/ae/aesm_service/data/white_list_cert_to_be_verify.bin $out/aesm/data/
+
+      # render: copy shared libraries
+      mkdir -p $out/lib64
+      cp ${build}/libsgx_uae_service.so $out/lib64/
+      cp ${build}/libsgx_epid.so $out/lib64/libsgx_epid.so.1
+      cp ${build}/libsgx_launch.so $out/lib64/libsgx_launch.so.1
+      cp ${build}/libsgx_quote_ex.so $out/lib64/libsgx_quote_ex.so.1
+      cp ${build}/libsgx_urts.so $out/lib64/
+      cp ${build}/libsgx_enclave_common.so $out/lib64/libsgx_enclave_common.so.1
+
+      # install provided udev rules
+      mkdir -p $out/etc/udev/rules.d
+      cp linux/installer/common/libsgx-enclave-common/91-sgx-enclave.rules $out/etc/udev/rules.d/
+      cp linux/installer/common/sgx-aesm-service/92-sgx-provision.rules $out/etc/udev/rules.d/
+    '';
+
+  # PSW "depends on itself" because it needs SDK headers to compile
+  buildInputs = [
+    curl
+    openssl
+  ] ++ (lib.optional (type == "psw") pkgs.intel-sgx-sdk);
+
+  nativeBuildInputs = [
+    autoconf
+    automake
+    cmake
+    libtool
+    ocaml
+    ocamlPackages.ocamlbuild
+    perl
+    protobuf
+    which
+    file
+    git
+    gnum4
+    texinfo
+    bison
+    flex
+  ];
+
+  makeFlags = [ type ]
+    # include the path to the SDK for PSW builds which depend on it
+    ++ lib.optional (type == "psw") "SGX_SDK=${pkgs.intel-sgx-sdk}/usr/share/sgxsdk"
+    ++ lib.optional debug "DEBUG=1";
+
+  # must disable hardening for debug builds
+  hardeningDisable = lib.optional debug "fortify";
+
+  dontUseCmakeConfigure = true;
+
+  meta = with lib; {
+    description = "Intel SGX for Linux";
+    homepage = "https://01.org/intel-softwareguard-extensions";
+    license = licenses.bsd3;
+    maintainers = with maintainers; [ citadelcore ];
+    platforms = platforms.linux;
```

The packages also only work on `x86_64`:

```suggestion
    platforms = [ "x86_64-linux" ];
```